instrumentation/spring/spring-ws-2.0/javaagent/src/test/java/test/boot/SpringWsTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test.boot;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+import static io.opentelemetry.semconv.ExceptionAttributes.EXCEPTION_MESSAGE;
+import static io.opentelemetry.semconv.ExceptionAttributes.EXCEPTION_STACKTRACE;
+import static io.opentelemetry.semconv.ExceptionAttributes.EXCEPTION_TYPE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.collect.ImmutableMap;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpServerUsingTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.incubating.CodeIncubatingAttributes;
+import io.opentelemetry.test.hello_web_service.HelloRequest;
+import io.opentelemetry.test.hello_web_service.HelloRequestSoapAction;
+import io.opentelemetry.test.hello_web_service.HelloRequestWsAction;
+import io.opentelemetry.test.hello_web_service.HelloResponse;
+import java.net.URISyntaxException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.oxm.jaxb.Jaxb2Marshaller;
+import org.springframework.util.ClassUtils;
+import org.springframework.ws.client.core.WebServiceMessageCallback;
+import org.springframework.ws.client.core.WebServiceTemplate;
+import org.springframework.ws.soap.addressing.client.ActionCallback;
+import org.springframework.ws.soap.client.SoapFaultClientException;
+import org.springframework.ws.soap.client.core.SoapActionCallback;
+
+public class SpringWsTest extends AbstractHttpServerUsingTest<ConfigurableApplicationContext> {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing =
+      HttpServerInstrumentationExtension.forAgent();
+
+  private ConfigurableApplicationContext context;
+
+  private static final Jaxb2Marshaller marshaller = new Jaxb2Marshaller();
+
+  @BeforeAll
+  void setupOptions() {
+    startServer();
+  }
+
+  @AfterAll
+  void cleanup() {
+    cleanupServer();
+  }
```

Comment:
the methods in the super class are almost the same, do you really need these here?

Reply:
I use `AbstractHttpServerUsingTest` instead of `AbstractHttpServerTest` thus I need to use beforeAll and afterAll junit annotations for starting cleanup and startServer methods directly in SpringWsTest

Reply:
thanks, that explains it